### PR TITLE
Enable hyperlinks in dataset description

### PR DESCRIPTION
**Pull request by rbernotas**
### Feature or Bugfix

- Feature

### Detail
- Enables conversion of hyperlinks in dataset description text.

### Relates
- https://github.com/data-dot-all/dataall/issues/1590

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?  Dataset description field is parsed for hyperlinks
  - Is the input sanitized? Yes, and additional sanitization can take place in linkMarkup.js
  - What precautions are you taking before deserializing the data you consume?  N/A
  - Is injection prevented by parametrizing queries?  N/A
  - Have you ensured no `eval` or similar functions are used? Yes
- Does this PR introduce any functionality or component that requires authorization? No
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms? N/A
  - Are you logging failed auth attempts? N/A
- Are you using or adding any cryptographic features? N/A
  - Do you use a standard proven implementations? N/A
  - Are the used keys controlled by the customer? Where are they stored? N/A
- Are you introducing any new policies/roles/users? N/A
  - Have you used the least-privilege principle? How? N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/utils/helpers/linkMarkup.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+export const createLinkMarkup = (text, color) => {
```
Comment (dlpzx) [inline frontend/src/utils/helpers/linkMarkup.js:1]: nit: Is the color ever going to change? If not, could we get it from the theme directly in the linkMarkup util?

**Review by dlpzx — APPROVED** [review]
Hi @rbernotas, left some minor comments, but the functionality looks good. Thanks for the PR

File: frontend/src/utils/helpers/linkMarkup.js (hunk 1 above)
Comment (rbernotas) [reply frontend/src/utils/helpers/linkMarkup.js:1]: I did look into this.  The color does need to change when you swap from light to dark theme in the settings.  But eslint doesn't like it when the useTheme hook is called within linkMarkup as createLinkMarkup isn't a React component or custom hook.  I looked into wrapping the function in a component, but then calling that started to get a little hacky just for the purpose of avoiding the param.

**Comment by rbernotas** [comment]
integration tests had a failure in test_submit_share_extension_request_with_auto_approval.  The error doesn't look related to this code change... is this a known issue?

File: frontend/src/utils/helpers/linkMarkup.js (hunk 1 above)
Comment (dlpzx) [reply frontend/src/utils/helpers/linkMarkup.js:1]: Understood, seems like the solution is more complex than the problem itself. I think we can leave it as is.

**Comment by dlpzx** [comment]
> integration tests had a failure in test_submit_share_extension_request_with_auto_approval. The error doesn't look related to this code change... is this a known issue?

Yes, there was a small issue on the expiration date calculation for the last day of a month. The fix is in this [PR](https://github.com/data-dot-all/dataall/pull/1594). I re-run the failed job for this PR because today, 1st of October, we don't run into the issue

**Pull request merged**